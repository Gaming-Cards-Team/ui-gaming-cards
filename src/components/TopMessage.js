import React from "react";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import {
  VIDEO_ALTERNATIVE_TEXT,
  MESSAGE_FOR_OTHER_CARDS,
  HERE,
  HOW_TO_PLAY,
} from "../util/constant";
import youtubeVideoImage from "../assets/images/youtubeVideoImage.png";

const topMessageStyle = {
  fontSize: "larger",
  background: "black",
  width: "100%",
  color: "white",
  padding: "10px 30px",
};
const messageStyle = {
  padding: "10px 0px",
};
const videoTagStyle = {
  textAlign: "right",
};
const linkStyle = {
  color: "#2d963f",
  fontWeight: "bold",
};

const TopMessage = () => (
  <div style={topMessageStyle}>
    <Row>
      <Col xs={9} style={messageStyle}>
        {MESSAGE_FOR_OTHER_CARDS}
        <a style={linkStyle} href="{#}">
          {HERE}
        </a>
      </Col>
      <Col xs={3} style={videoTagStyle}>
        <a style={linkStyle} href="https://www.youtube.com/watch?v=3gPu9pp2WOM">
          {HOW_TO_PLAY}
          <img
            data-testid="youtubeVideo"
            height="48"
            src={youtubeVideoImage}
            alt={VIDEO_ALTERNATIVE_TEXT}
          />
        </a>
      </Col>
    </Row>
  </div>
);

export default TopMessage;
