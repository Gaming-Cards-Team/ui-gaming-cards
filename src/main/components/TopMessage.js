import React from "react";
import Container from "react-bootstrap/Container";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";
import youtubeVideoImage from "../resources/youtubeVideoImage.png";

const topMessageStyle = {
  fontSize: "larger",
  background: "black",
  width: "100%",
  color: "white",
  padding: "10px 30px",
};
const messageStyle = {
  padding: "10px 0px",
};
const videoTagStyle = {
  textAlign: "right",
};
const linkStyle = {
  color: "#2d963f",
  fontWeight: "bold",
};

const TopMessage = () => (
  <div style={topMessageStyle}>
    <Row>
      <Col xs={9} style={messageStyle}>
        Are you looking for cards for other game cards? check our catalog
        <a style={linkStyle} href="https://www.youtube.com/"> here</a>
      </Col>
      <Col xs={3} style={videoTagStyle}>
        <a style={linkStyle} href="https://www.youtube.com/watch?v=3gPu9pp2WOM">
          How to play
          <img
            data-testid="youtubeVideo"
            height="48"
            src={youtubeVideoImage}
            alt="Level Up Logo"
          />
        </a>
      </Col>
    </Row>
  </div>
);

export default TopMessage;
