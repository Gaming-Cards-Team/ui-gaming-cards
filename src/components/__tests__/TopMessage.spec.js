import React from "react";
import TopMessage from "../TopMessage";
import renderer from "react-test-renderer";

describe("Top message component", () => {
  it("should render successfully", () => {
    const component = renderer.create(<TopMessage />).toJSON();

    expect(component).toMatchSnapshot();
  });
});