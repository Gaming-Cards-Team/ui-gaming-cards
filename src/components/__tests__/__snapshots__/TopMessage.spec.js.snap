// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Top Message component Should validate the component 1`] = `
<div
  style={
    Object {
      "background": "black",
      "color": "white",
      "fontSize": "larger",
      "padding": "10px 30px",
      "width": "100%",
    }
  }
>
  <div
    className="row"
  >
    <div
      className="col-9"
      style={
        Object {
          "padding": "10px 0px",
        }
      }
    >
      Are you looking for cards for other game cards? check our catalog 
      <a
        href="{#}"
        style={
          Object {
            "color": "#2d963f",
            "fontWeight": "bold",
          }
        }
      >
        here
      </a>
    </div>
    <div
      className="col-3"
      style={
        Object {
          "textAlign": "right",
        }
      }
    >
      <a
        href="https://www.youtube.com/watch?v=3gPu9pp2WOM"
        style={
          Object {
            "color": "#2d963f",
            "fontWeight": "bold",
          }
        }
      >
        How to play 
        <img
          alt="Video Logo"
          data-testid="youtubeVideo"
          height="48"
          src="youtubeVideoImage.png"
        />
      </a>
    </div>
  </div>
</div>
`;
